### PR TITLE
Add info about feature freeze period

### PR DESCRIPTION
**Pull request by OskarStark**
Follows
* https://github.com/symfony/symfony/pull/52425#issuecomment-1792244943

### CODE REVIEW

**Comment by BenMorel** [comment]
Thank you, @OskarStark!

- Could the "short timeframe" be defined, like `x` weeks before release?
- Could this be reflect on the [Symfony Releases Calendar](https://symfony.com/releases#symfony-releases-calendar)?

**Comment by OskarStark** [comment]
> Could the "short timeframe" be defined, like x weeks before release?

Not sure, as this is not a _hard_, more a _soft_ deadline, lets ask @nicolas-grekas and @fabpot 

> Could this be reflect on the [Symfony Releases Calendar](https://symfony.com/releases#symfony-releases-calendar)?

Not sure it's worth, but possibly yes. But once done, @carsonbot could use that information

**Comment by xabbuh** [comment]
We already document this as the "stabilization" period of two months here: https://symfony.com/doc/current/contributing/community/releases.html#development

**Comment by OskarStark** [comment]
I agree, but this does not really explain what that mean for the contributor

File: contributing/code/pull_requests.rst [hunk 1]
```diff
@@ -147,6 +147,15 @@ work:
     for the ``5.4`` branch, the PR will also be applied by the core team on
     all the ``6.x`` branches that are still maintained.
 
+.. warning::
```
Comment (94noni) [inline contributing/code/pull_requests.rst]: isnt "info" enough?
"warning" seem to strong and devs may be not opening new PR :s 

what about:

during a short period of time prior to the release of a new symfony major version, the symfony repository
is feature freeze. Your pull requests are still more than welcome nonetheless they must target the next major version

something like that ^^

File: contributing/code/pull_requests.rst [hunk 2]
```diff
@@ -147,6 +147,15 @@ work:
...
+
+    For a short timeframe before a new major version is released, we are in a
```
Comment (alexandre-daubois) [inline contributing/code/pull_requests.rst]: Is it only before a major version? I have the feeling that I already saw some feature freeze for minor version to stabilize the whole thing, but I may be wrong :thinking:

**Comment by fabpot** [comment]
I don't think that this is needed as this is already described here: https://symfony.com/doc/current/contributing/community/releases.html#development

Maybe adding a link to this paragraph would be enough.

**Comment by wouterj** [comment]
I agree that we should not duplicate the release cycle documentation, but I like mentioning it on this page. It's common to see contributors confused what they should do with their PR during this phase. What about replacing the strong warning with a more subtle paragraph like:

```rst
During the :ref:`stabilization phase <contributing-release-development>`, the development branch is in
feature freeze. Please help the community prepare for the new version release. If you want to submit a
new feature pull request, you should target the next version. For example, if ``6.3`` reached feature
freeze, new features should target ``6.4``. If the ``6.4`` branch does not yet exists, target ``6.3``
and rebase your pull requests once the branch is created.
```

(second sentence may be removed, I can't help myself pushing everyone to not focus on writing new features during these 2 months :stuck_out_tongue: )

**Comment by fabpot** [comment]
> I agree that we should not duplicate the release cycle documentation, but I like mentioning it on this page. It's common to see contributors confused what they should do with their PR during this phase. What about replacing the strong warning with a more subtle paragraph like:
> 
> ```
> During the :ref:`stabilization phase <contributing-release-development>`, the development branch is in
> feature freeze. Please help the community prepare for the new version release. If you want to submit a
> new feature pull request, you should target the next version. For example, if ``6.3`` reached feature
> freeze, new features should target ``6.4``. If the ``6.4`` branch does not yet exists, target ``6.3``
> and rebase your pull requests once the branch is created.
> ```
> 
> (second sentence may be removed, I can't help myself pushing everyone to not focus on writing new features during these 2 months 😛 )

That works for me

**Comment by OskarStark** [comment]
Thanks @wouterj, I used your proposal

File: contributing/code/pull_requests.rst [hunk 3]
```diff
@@ -147,6 +147,12 @@ work:
     for the ``5.4`` branch, the PR will also be applied by the core team on
     all the ``6.x`` branches that are still maintained.
 
+During the :ref:`stabilization phase <contributing-release-development>`, the development branch is in
+feature freeze. Please help the community prepare for the new version release. If you want to submit a
+new feature pull request, you should target the next version. For example, if ``6.3`` reached feature
+freeze, new features should target ``6.4``. If the ``6.4`` branch does not yet exists, target ``6.3``
```
Comment (xabbuh) [inline contributing/code/pull_requests.rst]:
```suggestion
freeze, new features should target ``6.4``. If the ``6.4`` branch does not yet exist, target ``6.3``
```

**Review by xabbuh — APPROVED** [review]

**Review by alexandre-daubois — APPROVED** [review]
Great!

**Pull request merged**